naming/src/main/java/org/jboss/as/naming/subsystem/NamingBindingAdd.java

```diff
@@ -234,7 +234,7 @@ public String getJndiViewInstanceValue() {
                 final ClassLoader cl = WildFlySecurityManager.getCurrentContextClassLoaderPrivileged();
                 try {
                     WildFlySecurityManager.setCurrentContextClassLoaderPrivileged(objectFactoryClassInstance.getClass().getClassLoader());
-                    return String.valueOf(getReference().getInstance());
+                    return (String) getReference().getInstance();
```

Comment:
This is not valid: String.valueOf(null) returns a String with the word "null". (String) null returns a null reference.